apps/meteor/client/views/marketplace/hooks/useAppsCountQuery.ts

```diff
@@ -0,0 +1,41 @@
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQueryClient, useQuery } from '@tanstack/react-query';
+import { useCallback } from 'react';
+
+type Variant = 'success' | 'warning' | 'danger';
+
+const getProgressBarValues = (numberOfEnabledApps: number, enabledAppsLimit: number): { variant: Variant; percentage: number } => ({
+	variant: 'success',
+	...(numberOfEnabledApps + 1 === enabledAppsLimit && { variant: 'warning' }),
+	...(numberOfEnabledApps >= enabledAppsLimit && { variant: 'danger' }),
+	percentage: Math.round((numberOfEnabledApps / enabledAppsLimit) * 100),
+});
+
+export const useAppsCountQuery = (context: 'private' | 'explore' | 'installed' | 'enterprise' | 'requested') => {
+	const getAppsCount = useEndpoint('GET', '/apps/count');
+
+	return useQuery(
+		['apps/count', { context }],
```

Comment:
useQuery turn object into string in order to provide memoize, but come on no need to use this feature here

Reply:
If I understood what Tiago and Felipe did here. This "dependency" is necessary to refetch the data and change the boolean verifications of this query upon changing from the "main marketplace pages" to the private apps page. The problem is that this is not the most optimized way to do it since it will re-run the query on every page change. Since we are too deep into the release delivery I propose we solve this problem on a future patch since the UX of this works just the optimization that is not the best.